### PR TITLE
Tools: autotest: add IRLock SITL test

### PR DESCRIPTION
**Pull request by khancyr**
Change takeoff to be in Loiter as we need the loiter controler to do precland correctly

### CODE REVIEW

**Comment by rishabsingh3003** [comment]
I don't understand the takeoff in "Loiter" change. Why do we need that?

**Comment by khancyr** [comment]
there are two issues : 
we takeoff in stabilize so we don't have the full check for loiter and precland use the loiter controler, so I don't want to have a failure because we land normally.
Currently, we got an issue on autotest where the takeoff in stabilize trigger the thrust loss check. It doesn't happen on Loiter, so that is double win.
but it shouldn't change the result

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -3044,52 +3044,54 @@ def test_motortest(self, timeout=60):
         self.end_subtest("Testing percentage output")
 
     def fly_precision_sitl(self):
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:3046]: Misnamed now.

Perhaps we should split this like we do Plane's airspeed testing  - have acommon subroutine called for each backend you want to test?

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: Tools/autotest/arducopter.py (hunk 1 above)
Comment (khancyr) [reply Tools/autotest/arducopter.py]: why it is misnamed ? both are SITL only backend ! Anyway, I have change the name

**Comment by davidbuzz** [comment]
i think this is mergable now. lgtm

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**